active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,199 @@
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+### Create a simple NumberBox
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="simple NumberBox" Header="Enter equation:" PlaceholderText="A + B * C" />
+```
+
+![](images/numberbox-placeholder-text.png)
+
+### Enable calculation support
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for simple calculations" 
+    Header="Enter equation:" 
+    PlaceholderText="A + B * C" 
+    AcceptsCalculation="True" />
+```
+
+![](images/numberbox-acceptscalculation.png)
+
+### Add increment and decrement stepping
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for items in order" 
+    StepSize="0.1"
+    UpDownButtonsEnabled="True"
+    HyperDragEnabled="True" 
+    HyperScrollEnabled="True" />
+```
+
+![](images/numberbox-updownbuttons.png)
+
+### Format display and output
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for solution components" 
+    Header="Enter a percentage:"
+    PlaceholderText="2/3" 
+    MinValue="0"
+    MaxValue="100" 
+    AreLeadingZerosTrimmed="False"
+    DecimalPrecision="5"
+    DoesInputRound=False" />
+```
+
+## Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+## API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+## API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+```c++ 
+runtimeclass NumberBoxValueChangedEventArgs
+{
+    String Text;
+    Double Value
+};
+
+unsealed runtimeclass NumberBox : Windows.UI.Xaml.Controls.TextBox
+{
+    NumberBox();
+    
+    Double Value;
+    
+    Boolean AcceptsCalculation;
+    Boolean UpDownButtonsEnabled;
+    Boolean HyperDragEnabled;
+    Boolean HyperScrollEnabled;
+    
+    Double StepSize;
+  
+    Boolean AreLeadingZerosTrimmed;
+    Double DecimalPrecision;
+    Boolean DoesInputRound;
+    
+    Double MinValue;
+    Double MaxValue;
+    
+
+    NumberBoxTemplateSettings TemplateSettings{ get; };
+
+    event Windows.Foundation.TypedEventHandler<NumberBox, NumberBoxValueChangedEventArgs> ValueChanged;
```

Comment:
What will trigger a `ValueChanged` event?
Will `Value` be updated (and this event triggered) three times if I entered "123" in the box?
Will the event be triggered when text changes to an invalid value?
Presumably the event won't be changed when the text changes from one invalid value to another. This will mean that multiple events won't be invoked while entering a calculation until the calculation produces something valid.

I believe the above will handle all appropriate cases and allow for appropriate validation beyond an individual control until #179 is fully implemented.

Reply:
Great question, @mrlacey, and thanks for your patience as I continue to catch up on my inbox. I have detailed the event behaviors here in the API Notes section: https://github.com/microsoft/microsoft-ui-xaml-specs/blob/user/savoyschuler/numberbox/active/NumberBox/NumberBox.md#behavioral-components

If anything looks out of place, please let me know!